mlx/ops.cpp

```diff
@@ -753,6 +753,41 @@ array repeat(const array& arr, int repeats, StreamOrDevice s) {
   return repeat(flatten(arr, s), repeats, 0, s);
 }
 
+array tile(
+    const array& arr,
+    std::vector<int> reps,
+    StreamOrDevice s /* = {} */) {
+  if (reps.size() < arr.ndim()) {
+    reps.insert(reps.begin(), arr.ndim() - reps.size(), 1);
+  }
+
+  std::vector<int> expand_shape = arr.shape();
+  int offset = 0;
+  for (int i = 0; i < arr.ndim(); i++) {
+    if (reps[i] != 1) {
+      expand_shape.insert(expand_shape.begin() + offset + i, 1);
+      offset += 1;
+    }
+  }
+  offset = 0;
+  std::vector<int> broad_shape = expand_shape;
+  for (int i = 0; i < arr.ndim(); i++) {
+    if (reps[i] != 1) {
+      broad_shape[i + offset] *= reps[i];
+      offset += 1;
+    }
+  }
```

Comment:
I may be being a pedant here but I think this can be simplified as follows:

```cpp
std::vector<int> expand_shape;
std::vector<int> broadcast_shape;
std::vector<int> final_shape;
for (int i=0; i < arr.ndim(); i++) {
  expand_shape.push_back(arr.shape(i));
  broadcast_shape.push_back(arr.shape(i));
  final_shape.push_back(arr.shape(i));
  if (reps[i] != 1) {
    expand_shape.push_back(1);
    broadcast_shape.push_back(reps[i]);
    final_shape.back() *= reps[i];
  }
}
```

Reply:
modified it to something similar to this